qutip/solver/result.py

```diff
@@ -894,78 +919,208 @@ def __repr__(self):
 
     def __add__(self, other):
         if not isinstance(other, MultiTrajResult):
-            return NotImplemented
+            raise NotImplementedError("Can only add two multi traj results")
         if self._raw_ops != other._raw_ops:
             raise ValueError("Shared `e_ops` is required to merge results")
         if self.times != other.times:
             raise ValueError("Shared `times` are is required to merge results")
 
         new = self.__class__(
-            self._raw_ops, self.options, solver=self.solver, stats=self.stats
+            self.e_ops, self.options, solver=self.solver, stats=self.stats
```

Comment:
`e_ops` are callable wrapped in `ExpectOp`, by passing them here, they get wrapped twice and raw_ops are no longer available.